packages/material-ui-lab/src/Slider/Slider.js

```diff
@@ -0,0 +1,549 @@
+    primary: theme.palette.primary.main,
+    secondary: theme.palette.grey[400],
+    focused: theme.palette.grey[500],
+    disabled: theme.palette.grey[400],
+  };
+
+  return {
+    /* Styles for wrapper container */
+    container: {
+      position: 'relative',
+      width: '100%',
+      margin: '10px 0',
+      padding: '6px 0',
+      cursor: 'pointer',
+      '-webkit-tap-highlight-color': 'transparent',
+      '&$disabled': {
+        cursor: 'no-drop',
+      },
+      '&$vertical': {
+        height: '100%',
+        margin: '0 10px',
+        padding: '0 6px',
+      },
+      '&$reverse': {
+        transform: 'scaleX(-1)',
+      },
+      '&$vertical$reverse': {
+        transform: 'scaleY(-1)',
+      },
+    },
+    /* Tracks styles */
+    track: {
+      position: 'absolute',
+      transform: 'translate(0, -50%)',
+      top: '50%',
+      height: 2,
+      '&$focused, &$activated': {
+        transition: 'none',
+        backgroundColor: colors.focused,
+      },
+      '&$disabled': {
+        backgroundColor: colors.secondary,
+      },
+      '&$vertical': {
+        transform: 'translate(-50%, 0)',
+        left: '50%',
+        top: 'initial',
+        width: 2,
+      },
+      '&$jumped': {
+        backgroundColor: colors.focused,
+      },
+    },
+    trackBefore: {
+      zIndex: 1,
+      left: 0,
+      backgroundColor: colors.primary,
+      transition: commonTransitions,
+      '&$focused, &$activated, &$jumped': {
+        backgroundColor: colors.primary,
+      },
+    },
+    trackAfter: {
+      right: 0,
+      backgroundColor: colors.secondary,
+      transition: commonTransitions,
+      '&$vertical': {
+        bottom: 0,
+      },
+    },
+    /* Thumb styles  */
+    thumb: {
+      position: 'absolute',
+      zIndex: 2,
+      transform: 'translate(-50%, -50%)',
+      width: 12,
+      height: 12,
+      borderRadius: '50%',
+      transition: commonTransitions,
+      backgroundColor: colors.primary,
+      '&$focused': {
+        boxShadow: `0px 0px 0px 9px ${fade(colors.primary, 0.16)}`,
+      },
+      '&$activated': {
+        width: 17,
+        height: 17,
+        transition: 'none',
+      },
+      '&$disabled': {
+        cursor: 'no-drop',
+        width: 9,
+        height: 9,
+        backgroundColor: colors.disabled,
+      },
+      '&$zero': {
+        border: `2px solid ${colors.disabled}`,
+        backgroundColor: 'transparent',
+      },
+      '&$focused$zero': {
+        border: `2px solid ${colors.focused}`,
+        backgroundColor: fade(colors.focused, 0.34),
+        boxShadow: `0px 0px 0px 9px ${fade(colors.focused, 0.34)}`,
+      },
+      '&$activated$zero': {
+        border: `2px solid ${colors.focused}`,
+      },
+      '&$jumped': {
+        width: 17,
+        height: 17,
+      },
+    },
+    focused: {},
+    activated: {},
+    disabled: {},
+    zero: {},
+    vertical: {},
+    reverse: {},
+    jumped: {},
+  };
+};
+
+const KEY_CODES = {
```

Comment:
This is already packages into the `keycode` dependency. No need for duplicating it.

Reply:
You can find many examples in the codebase.